services/datamanager/datasync/sync_test.go

```diff
@@ -198,63 +193,59 @@ func TestSensorUploadBinary(t *testing.T) {
-		// that data be uploaded to the cloud
-		tf, err := createTmpDataCaptureFile()
-		if err != nil {
-			t.Errorf("%s cannot create temporary file to be used for sensorUpload/fileUpload testing: %v",
-				tc.name, err)
-		}
-		defer os.Remove(tf.Name())
-
-		// First write metadata to file.
-		syncMetadata := v1.DataCaptureMetadata{
-			ComponentType:    componentType,
-			ComponentName:    componentName,
-			ComponentModel:   componentModel,
-			MethodName:       methodName,
-			Type:             v1.DataType_DATA_TYPE_BINARY_SENSOR,
-			MethodParameters: nil,
-			FileExtension:    datacapture.GetFileExt(v1.DataType_DATA_TYPE_BINARY_SENSOR, methodName, nil),
-			Tags:             tags,
-		}
-		if _, err := pbutil.WriteDelimited(tf, &syncMetadata); err != nil {
-			t.Errorf("%s cannot write protobuf struct to temporary file as part of setup for sensorUpload testing: %v",
-				tc.name, err)
-		}
-
-		// Write the data from the test cases into the files to prepare them for reading by the sensorUpload function.
-		if err := writeSensorData(tf, tc.toSend); err != nil {
-			t.Errorf("%s cannot write byte slice to temporary file as part of setup for "+
-				"sensorUpload/fileUpload testing: %v", tc.name, err)
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			// Create temp file to be used as examples of reading data from the files into buffers and finally to have
+			// that data be uploaded to the cloud
+			tmpDir, err := ioutil.TempDir("", "")
+			test.That(t, err, test.ShouldBeNil)
+			defer os.RemoveAll(tmpDir)
+			// First write metadata to file.
+			captureMD := v1.DataCaptureMetadata{
+				ComponentType:    componentType,
+				ComponentName:    componentName,
+				ComponentModel:   componentModel,
+				MethodName:       methodName,
+				Type:             v1.DataType_DATA_TYPE_BINARY_SENSOR,
+				MethodParameters: nil,
+				FileExtension:    datacapture.GetFileExt(v1.DataType_DATA_TYPE_BINARY_SENSOR, methodName, nil),
+				Tags:             tags,
+			}
+			f, err := datacapture.NewFile(tmpDir, &captureMD)
+			test.That(t, err, test.ShouldBeNil)
 
-		// Upload the contents from the created file.
-		// Register mock datasync service with a mock server.
-		logger, _ := golog.NewObservedTestLogger(t)
-		mockService := getMockService()
-		rpcServer := buildAndStartLocalServer(t, logger, mockService)
-		defer func() {
-			err := rpcServer.Stop()
+			// Write the data from the test cases into the files to prepare them for reading by the sensorUpload function.
+			for i := range tc.toSend {
+				err := f.WriteNext(tc.toSend[i])
+				test.That(t, err, test.ShouldBeNil)
+			}
+			err = f.Sync()
 			test.That(t, err, test.ShouldBeNil)
-		}()
 
-		conn, err := getLocalServerConn(rpcServer, logger)
-		test.That(t, err, test.ShouldBeNil)
-		client := NewClient(conn)
-		sut, err := NewManager(logger, partID, client, conn)
-		test.That(t, err, test.ShouldBeNil)
-		sut.Sync([]string{tf.Name()})
+			// Upload the contents from the created file.
```

Comment:
It definitely can be extracted, but not that easily because in each of these cases we set different mockService parameters so would need to pass those in. I'd rather not include it in this PR.